packages/cms/app.js

```diff
@@ -292,7 +294,12 @@ module.exports.getMultiSiteApp = (options) => {
     const resetConfigMw = async (req, res, next) => {
         let host = req.headers['x-forwarded-host'] || req.get('host');
         host = host.replace(['http://', 'https://'], ['']);
-        await fetchAllSites(req, res);
+        try {
+            await fetchAllSites(req, res);
+        } catch (e) {
+            console.error('An error occurred fetching the site config:', e);
+            if (res) return res.status(500).json({error: 'An error occured fetching the sites data: ' + e});
+        }
         req.forceRestart = true;
         next();
```

Comment:
We have to do an early return because we can only return one response to each request. Is the req.forceRestart doing something important?

The next() loads the next middleware, that middleware can emit another response.